src/undomanager.js

```diff
@@ -46,6 +48,7 @@ define([
         appendMug: function (mug, previousMug, position) {
             this.undoStack = this.undoStack.concat([[mug, previousMug, position]]);
             toggleAlert(this.undoStack);
+            windowManager.adjustToWindow(mug.form.vellum);
```

Comment:
Another way to call this that would avoid the need to extract it to a module-level function is `mug.form.vellum.data.windowManager.adjustToWindow()`. Rather than (ab)using that long method path, I'd add a new core method `vellum.adjustToWindow()` that calls `this.data.windowManager.adjustToWindow()`.

And since `undomanager.appendMug()` is only called in one place, what do you think of pushing this call into [`form._addToUndoManager`](https://github.com/dimagi/Vellum/blob/6db614856293b222c5d3928d12a7a0429037524e/src/form.js#L888), like so:
```js
...
this.undomanager.appendMug(mug, previousSibling, position);
this.vellum.adjustToWindow()
...
```
The refactor mentioned at the top of this comment is still necessary, but that would eliminate some indirection.